lib/logqueue.c

```diff
@@ -168,18 +168,60 @@ log_queue_check_items(LogQueue *self, gint *timeout, LogQueuePushNotifyFunc para
   return TRUE;
 }
 
-void
-log_queue_set_counters(LogQueue *self, StatsCounterItem *queued_messages, StatsCounterItem *dropped_messages,
-                       StatsCounterItem *memory_usage)
+static void
+_reset_counters(LogQueue *self)
 {
-  self->queued_messages = queued_messages;
-  self->dropped_messages = dropped_messages;
-  self->memory_usage = memory_usage;
   stats_counter_set(self->memory_usage,
                     self->memory_usage_qout_initial_value + self->memory_usage_overflow_initial_value);
   stats_counter_set(self->queued_messages, log_queue_get_length(self));
 }
 
+static void
+_register_common_counters(LogQueue *self, gint stats_level, const StatsClusterKey *sc_key)
+{
+  gboolean need_to_reset_counters = TRUE;
+
+  stats_register_counter(stats_level, sc_key, SC_TYPE_QUEUED, &self->queued_messages);
+
+  if (stats_check_level(STATS_LEVEL1))
+    need_to_reset_counters = !stats_contains_counter(sc_key, SC_TYPE_MEMORY_USAGE);
```

Comment:
We could go this check into _reset_counters.

Reply:
No, it couldn't.
Why?
Because at that point the counter is registered... so it would return `TRUE`  (of course when stats-level is >= 1).